### ISSUE TITLE
[Android] no gradient when `locations` param specified

### ISSUE THREAD

**Issue by knowbody** [brentvatne/react-native-linear-gradient#81]
Seems like specifying `locations` stops displaying gradient on Android.

example code snippet:

```js
<LinearGradient
  locations={[0.3, 0.6, 1]}
  colors={['rgba(0, 0, 0, 0)', 'rgba(49, 63, 116, 0.8)']}
  style={styles.container}
>
  {/* ... */}
</LinearGradient>
```

it works fine on iOS, on Android the gradient is not displayed at all. Removing `locations` shows the gradient, but with the default locations.

**Edit:**

Seems that it is happening because the *middle* value is specified. it work just fine with two values (e.g. `locations={[0, 1]}` or `locations={[0.3, 0.8]}`)

**Comment by dgladkov** [comment]
Each element in the locations array corresponds to the element with the same index in the color array, so they should have the same size. However iOS's `CAGradientLayer` simply ignores locations that do not have a corresponding color (if you remove 1 in your example the gradient will look exactly the same on iOS) while Android's `LinearGradient` throws exception, so currently we have additional check to ensure that colors and locations arrays have the same size that prevents rendering, that's the reason for inconsistent behavior

So I think we should slice the locations array on Android to the size of colors array to have consistent behavior and also display a warning when colors and locations arrays have different length

**Comment by knowbody** [comment]
that would be great! thank you

**Comment by dgladkov** [comment]
Fixed in 1.5.8

**Comment by dgladkov** [comment]
Please update to 1.5.9, 1.5.8 introduced a new problem for Android when locations prop was missing
